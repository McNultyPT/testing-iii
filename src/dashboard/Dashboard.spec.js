import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import Dashboard from './Dashboard.js';
import Controls from '../controls/Controls.js'

describe('<Dashboard />', () => {
    it('renders without crashing', () => {
        render(<Dashboard />)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<Dashboard />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    describe('toggleLocked()', () => {
        it('it toggles when clicked', () => {
            const { getByTestId } = render(<Controls />);

            const button1 = getByTestId('toggleLock');
            const button2 = getByTestId('toggleOpen');
            expect(getByTestId('toggleLock')).toHaveTextContent('Lock Gate');

            fireEvent.click(button2);
            fireEvent.click(button1);
            expect(getByTestId('toggleLock')).toHaveTextContent('Unlock Gate');
        });
    });

    // describe('toggleClosed()', () => {
    //     it('it toggles when clicked', () => {
    //         const { getByTestId } = render(<Controls closed={false} />);

    //         const button1 = getByTestId('toggleLock');
    //         const button2 = getByTestId('toggleOpen');
    //         expect(getByTestId('toggleOpen')).toHaveTextContent('Close Gate');

    //         fireEvent.click(button2);
    //         expect(getByTestId('toggleOpen')).toHaveTextContent('Open Gate');
    //     });
    // });
});
