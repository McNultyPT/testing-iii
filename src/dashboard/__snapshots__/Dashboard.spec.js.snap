// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="locked"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="closed"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="toggleLock"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="toggleOpen"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
